### PR TITLE
VSCode addition to .gitignore

### PR DESCRIPTION
**Pull request by fabianhough**
I'm currently using vscode; it has a .vscode folder that it places within the repo.  I didn't want to forget down the road.  I hope this is okay, first time really modifying .gitignore.  I understand if this isn't standard practice!

### CODE REVIEW

**Comment by AKuederle** [comment]
No this is totally fine! but check my comment in the review

File: .gitignore [hunk 1]
```diff
@@ -99,3 +99,11 @@ ENV/
 
 # mypy
 .mypy_cache/
+
+### VisualStudioCode ###
+.vscode/*
+!.vscode/settings.json
```
Comment (AKuederle) [inline .gitignore]: All the lines starting with a `!` are included again. This means, that the task, launch, and extesnion file will still be added.
This is something you might want to do if you work on a project alone and want to  sync part of your settings between PCs, but for a multi people project you would want to exclude them as well.
So please remove the 4 exception lines

**Comment by fabianhough** [comment]
Okay!  What about the .history?  Is that going to have untintended consequences?

**Comment by AKuederle** [comment]
I wasn't aware that vscode creates a .history file. But I am also sure that no component of our library will use it. So it should be safe to ignore it.

**Pull request merged**